src/App.js

```diff
@@ -1,55 +1,63 @@
 import React from 'react'
+import {debounce} from './helpers'
+
+// API
 import FakeSearchAPI from './fake-api/search'
+
+// COMPONENT
 import { PAutoComplete } from './components/widgets'
+
+// STYLE
 import './App.css'
 
 class App extends React.Component {
   constructor(props) {
     super(props)
+
     this.state = {
       keyword: '',
       results: [],
+      loading: false,
     }
   }
-  updateKeyword(keyword) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          keyword,
-        },
-        () => resolve(keyword)
-      )
-    })
-  }
-  updateResult(results) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          results,
-        },
-        () => resolve()
-      )
-    })
-  }
-  async search(keyword) {
-    try {
-      // Update keyword input
-      this.updateKeyword(keyword)
-      // Get all the items which start with `keyword`
-      const results = await FakeSearchAPI.search(keyword)
-      // Update suggestion list
+
+  updateKeyword = (keyword) => this.setState({keyword})
+
+  updateResult = (results) => this.setState({results})
+
+  search = debounce(async () => {    
+    // Get all the items which start with `keyword`
+
+    this.setState({ loading: true })
+
+    const results = await FakeSearchAPI.search(this.state.keyword)
+
+    // Update suggestion list
+
+    if (results && Array.isArray(results)) {
+      this.setState({ loading: false })
```

Comment:
You only set loading when there is `results` and the `results` is an Array. Mean that if the `results` is not an Array or is null, then it still remain loading

Reply:
Yes, the loading status should be revoked before the if condition, my bad.